pkg/cmd/container/stats.go

```diff
@@ -0,0 +1,432 @@
+		}
+		cStats.mu.Unlock()
+
+		if !firstTick {
+			// print header for every tick
+			if options.Format == "" || options.Format == "table" {
+				fmt.Fprintln(w, "CONTAINER ID\tNAME\tCPU %\tMEM USAGE / LIMIT\tMEM %\tNET I/O\tBLOCK I/O\tPIDS")
+			}
+		}
+
+		for _, c := range ccstats {
+			if c.ID == "" {
+				continue
+			}
+			rc := statsutil.RenderEntry(&c, options.NoTrunc)
+			if !firstTick {
+				if tmpl != nil {
+					var b bytes.Buffer
+					if err := tmpl.Execute(&b, rc); err != nil {
+						break
+					}
+					if _, err = fmt.Fprintln(options.Stdout, b.String()); err != nil {
+						break
+					}
+				} else {
+					if _, err := fmt.Fprintf(w, "%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\n",
+						rc.ID,
+						rc.Name,
+						rc.CPUPerc,
+						rc.MemUsage,
+						rc.MemPerc,
+						rc.NetIO,
+						rc.BlockIO,
+						rc.PIDs,
+					); err != nil {
+						break
+					}
+				}
+			}
+		}
+		if f, ok := w.(formatter.Flusher); ok {
+			f.Flush()
+		}
+
+		if len(cStats.cs) == 0 && !showAll {
+			break
+		}
+		if options.NoStream && !firstTick {
+			break
+		}
+		select {
+		case err, ok := <-closeChan:
+			if ok {
+				if err != nil {
+					return err
+				}
+			}
+		default:
+			// just skip
+		}
+		firstTick = false
+	}
+
+	return err
+}
+
+func collect(ctx context.Context, globalOptions types.GlobalCommandOptions, s *statsutil.Stats, waitFirst *sync.WaitGroup, id string, noStream bool) {
+	logrus.Debugf("collecting stats for %s", s.Container)
+	var (
+		getFirst = true
+		u        = make(chan error, 1)
+	)
+
+	defer func() {
+		// if error happens and we get nothing of stats, release wait group whatever
+		if getFirst {
+			getFirst = false
+			waitFirst.Done()
+		}
+	}()
+	client, ctx, cancel, err := clientutil.NewClient(ctx, globalOptions.Namespace, globalOptions.Address)
+	if err != nil {
+		s.SetError(err)
+		return
+	}
+	defer cancel()
+	container, err := client.LoadContainer(ctx, id)
+	if err != nil {
+		s.SetError(err)
+		return
+	}
+
+	go func() {
+		previousStats := new(statsutil.ContainerStats)
+		firstSet := true
+		for {
+			//task is in the for loop to avoid nil task just after Container creation
+			task, err := container.Task(ctx, nil)
+			if err != nil {
+				u <- err
+				continue
+			}
+
+			//labels is in the for loop to avoid nil labels just after Container creation
+			clabels, err := container.Labels(ctx)
+			if err != nil {
+				u <- err
+				continue
+			}
+
+			metric, err := task.Metrics(ctx)
+			if err != nil {
+				u <- err
+				continue
+			}
+			anydata, err := typeurl.UnmarshalAny(metric.Data)
+			if err != nil {
+				u <- err
+				continue
+			}
+
+			netNS, err := containerinspector.InspectNetNS(ctx, int(task.Pid()))
+			if err != nil {
+				u <- err
+				continue
+			}
+
+			// when (firstSet == true), we only set container stats without rendering stat entry
+			statsEntry, err := setContainerStatsAndRenderStatsEntry(previousStats, firstSet, anydata, int(task.Pid()), netNS.Interfaces)
```

Comment:
```suggestion
			statsEntry, err := setContainerStatsThenRenderStatsEntry(previousStats, firstSet, anydata, int(task.Pid()), netNS.Interfaces)
```

Reply:
"And" seems more common to me than "Then" in my impression. Any specific reasons of preferring "Then"?

(This is out of the scope of the current refactoring goal but I'm good to quickly include it if it makes sense)

Reply:
what happens under vu: the first iteration we define only `ContainerStats`, then the second iteration we define a new `ContainerStats` and render the statistics by making differences. My idea was to reflect this strategy in the title of the function but we can keep it because you don't feel the need @AkihiroSuda @ningziwen

Reply:
To me, for the meanings, "and" and "then" should be interchangable in Go naming. 

Even In daily usage, they seem very similar in most scenarios.
"I opened the door, and entered the room"
"I opened the door, then entered the room"

I'm leaning to keeping the more common one "and" but don't have strong preference.